addons/sys_core/XEH_preInit.sqf

```diff
@@ -148,5 +126,21 @@ for "_i" from 1 to (_m/2) do {
     };
 };
 
+addMissionEventHandler ["ExtensionCallback", {
+    params ["_name", "_function", "_data"];
+    if (_name != "ACRE_TR") exitWith {};
+    (parseSimpleArray _data) params ["_id", "_args"];
+    TRACE_3("ExtensionCallback",_function,_id,_data);
+
+    private _callBack = GVAR(threadedExtCalls) select _id;
+
+    if (IS_ARRAY(_callBack)) then {
+        if !(_args isEqualTo []) then {
+            _args = _args select 0;
+        };
+        [_callBack select 0, _args] call (_callBack select 1);
```

Comment:
```suggestion
        [_callBack select 0, _args] call (_callBack select 1);
        GVAR(threadedExtCalls) set [_id, nil];
```
Would it make sense to clear the entry after use?
`count str acre_sys_signal_fnc_handleSignalReturn` is 5368 and the whole func is copied each time
so after several thousand calls this will start to take up a non-trivial amount of memory

could also store ID in each element and pop off when used
```
(GVAR(threadedExtCalls) param [0,[-1]]) params ["_firstId", "_firstArgs", "_firstFunc"];
if (_id != _firstId) exitWith {ERROR...};
GVAR(threadedExtCalls) deleteAt 0;
```

or use the _function param to invoke the handler
```
if (_function == "process_signal") then { ... call EFUNC(...,handleSignalReturn)};
```

Reply:
I don't think this will break anything. But tagging @Soldia1138 and @jonpas for another opinion.

Reply:
It should work, yes. Everything's set to missionNamespace within `acre_sys_signal_fnc_handleSignalReturn`